bpfman-operator/apis/v1alpha1/cluster_uprobe_program_types.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClUprobeProgramInfo contains the information for the uprobe program
+type ClUprobeProgramInfo struct {
+	// links in the list of points to which the program should be attached.  The list items
+	// are optional and may be udated after the bpf program has been loaded
+	// +optional
+	// +kubebuilder:default:={}
+	Links []ClUprobeAttachInfo `json:"links"`
+}
+
+type ClUprobeAttachInfo struct {
+	// function to attach the uprobe to.
+	// +optional
+	Function string `json:"function"`
+
+	// offset added to the address of the function for uprobe.
+	// +optional
+	// +kubebuilder:default:=0
+	Offset uint64 `json:"offset"`
+
+	// target is the Library name or the absolute path to a binary or library.
+	Target string `json:"target"`
```

Comment:
Any validations you can put in place to ensure a valid value is provided by a user at admission time?